### PR TITLE
feat(api): importing lpaq (ASI-20)

### PR DESCRIPTION
**Pull request by fabfab-kc**
Imports the LPA questionnaire submission through the Service Bus.
- Refactored DB tables
- Applied mapping through web and API
- Improve DB queries
- General improvements

## Issue ticket number and link
[ASI-20](https://pins-ds.atlassian.net/browse/ASI-20)

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/web/src/server/lib/mappers/appeal.mapper.js [hunk 1]
```diff
@@ -725,12 +725,12 @@ export async function initialiseAndMapAppealData(
...
 								text: 'Yes',
 								value: 'yes',
-								checked: appealDetails.isAffectingNeighbouringSites
+								checked: appealDetails.neighbouringSites?.length > 0
 							},
 							{
 								text: 'No',
 								value: 'no',
-								checked: !appealDetails.isAffectingNeighbouringSites
+								checked: !appealDetails.neighbouringSites?.length
```
Comment (AlexMoulin) [inline appeals/web/src/server/lib/mappers/appeal.mapper.js:733]: Should be !appealDetails.neighbouringSites

File: appeals/web/src/server/lib/mappers/__tests__/mappers.test.js [hunk 2]
```diff
@@ -88,6 +88,7 @@ describe('lpaQuestionnaire-mapper', () => {
 	beforeAll(async () => {
 		currentRoute = 'testroute/';
 		validMappedData = await initialiseAndMapLPAQData(
+			// @ts-ignore
```
Comment (AlexMoulin) [inline appeals/web/src/server/lib/mappers/__tests__/mappers.test.js:91]: Shouldn't be needed?

File: appeals/api/src/server/tests/integrations/mocks.js [hunk 3]
```diff
@@ -1,134 +1,109 @@
...
-		caseReference: '6000526',
-		LPACode: 'Q9999',
-		isAppealTypeAppropriate: true,
-		doesTheDevelopmentAffectTheSettingOfAListedBuilding: true,
-		affectedListedBuildings: [
+	casedata: {
+		caseReference: '6000000',
+		lpaQuestionnaireSubmittedDate: new Date(2024, 5, 1).toISOString(),
+		lpaStatement: 'cupidatat ipsum eu culpa',
+		siteAccessDetails: ['Here it is'],
```
Comment (AlexMoulin) [inline appeals/api/src/server/tests/integrations/mocks.js:63]: It's an array now?

File: appeals/api/src/server/tests/integrations/mocks.js [hunk 4]
```diff
@@ -1,134 +1,109 @@
...
-		],
-		appellantCostsAppliedFor: false
+		lpaCode: 'Q9999',
+		nearbyCaseReferences: ['1000000'],
+		neighbouringSiteAddresses: [],
+		originalDevelopmentDescription: 'A test description',
+		ownersInformed: true,
+		ownsAllLand: true,
+		ownsSomeLand: true,
+		siteAccessDetails: ['Come and see'],
```
Comment (AlexMoulin) [inline appeals/api/src/server/tests/integrations/mocks.js:25]: It's now an array?

File: appeals/api/src/server/openapi-types.ts [hunk 5]
```diff
@@ -34,145 +34,140 @@ export interface ValidateDate {
...
-		/** @example true */
-		hasRepresentationsFromOtherParties?: boolean;
-		/** @example true */
-		doesSiteRequireInspectorAccess?: boolean;
+		/** @example "2024-05-31T23:00:00.000Z" */
+		lpaQuestionnaireSubmittedDate?: string;
+		/** @example "cupidatat ipsum eu culpa" */
+		lpaStatement?: string;
+		/** @example ["Here it is"] */
+		siteAccessDetails?: string[];
```
Comment (AlexMoulin) [inline appeals/api/src/server/openapi-types.ts:138]: Is there a reason that siteAccessDetails was changed to array? If justified, why not do the same for siteSafetyDetails?

File: appeals/api/src/server/openapi-types.ts [hunk 6]
```diff
@@ -34,145 +34,140 @@ export interface ValidateDate {
...
-		}[];
-		/** @example false */
-		appellantCostsAppliedFor?: boolean;
+		ownersInformed?: boolean;
+		/** @example true */
+		ownsAllLand?: boolean;
+		/** @example true */
+		ownsSomeLand?: boolean;
+		/** @example ["Come and see"] */
+		siteAccessDetails?: string[];
```
Comment (AlexMoulin) [inline appeals/api/src/server/openapi-types.ts:82]: See below

**Review by AlexMoulin — CHANGES_REQUESTED** [review]
Some comments

File: appeals/api/src/server/tests/integrations/mocks.js (hunk 3 above)
Comment (fabfab-kc) [reply appeals/api/src/server/tests/integrations/mocks.js:63]: yes, comes and goes as an array, but there will only be 2 items (the appellant and lpa answers)

File: appeals/api/src/server/tests/integrations/mocks.js (hunk 4 above)
Comment (fabfab-kc) [reply appeals/api/src/server/tests/integrations/mocks.js:25]: as above

File: appeals/api/src/server/openapi-types.ts (hunk 5 above)
Comment (fabfab-kc) [reply appeals/api/src/server/openapi-types.ts:138]: it is!

File: appeals/web/src/server/lib/mappers/appeal.mapper.js (hunk 1 above)
Comment (fabfab-kc) [reply appeals/web/src/server/lib/mappers/appeal.mapper.js:733]: I left that in to add an extra bool in the UI, but can remove the whole row...

**Review by glpins — DISMISSED** [review]

**Review by AlexMoulin — APPROVED** [review]
LGTM

**Pull request merged**